src/MultiInstanceFile.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of vfsStream.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace bovigo\vfs;
+
+use const SEEK_SET;
+
+/**
+ * Decorator for vfsStreamFile to allow multiple instances of a file to be open.
+ *
+ * It works by tracking and restoring the position in the file for each specific
+ * instance created, even though the underlying file is shared.
+ *
+ * @internal
+ */
+final class MultiInstanceFile
```

Comment:
I'm ok with naming it whatever people think is best. I only named it `MultiInstanceFile` because I thought it described what it did, allowing multiple instances of the same file to be opened/interacted with without interference on other instances of that same file.

This class is only used during the `stream_open` stage so I guess `OpenedFile` makes sense, too.

If nobody says otherwise, I'll rename it to that.

Reply:
Renamed